contracts/supplementary/rebalancing-manager/RebalancingTokenManager.sol

```diff
@@ -0,0 +1,456 @@
+                _btcPrice,
+                _ethPrice,
+                naturalUnit,
+                units
+            );          
+        }
+
+        return(naturalUnit, nextSetDollarAmount, units);
+    }
+
+    /*
+     * Determine units and naturalUnit of nextSet to propose
+     *
+     * @param  _currentSetDollarAmount      The 18 decimal value of one currenSet
+     * @param  _nextSetDollarAmount         The 18 decimal value of one nextSet
+     * @return uint256                      The auctionStartPrice for rebalance auction
+     * @return uint256                      The auctionPivotPrice for rebalance auction
+     */
+    function calculateAuctionPriceParameters(
+        uint256 _currentSetDollarAmount,
+        uint256 _nextSetDollarAmount
+    )
+        private
+        returns (uint256, uint256)
+    {
+        // Determine fair value of nextSet/currentSet and put in terms of auction library price divisor
+        uint256 fairValue = _nextSetDollarAmount.mul(AUCTION_LIB_PRICE_DIVISOR).div(_currentSetDollarAmount);
+        // Calculate how much one percent slippage from fair value is
+        uint256 onePercentSlippage = fairValue.div(100);
+
+        // Calculate how many 30 minute periods are in auctionTimeToPivot
+        uint256 thirtyMinutePeriods = auctionTimeToPivot.div(THIRTY_MINUTES_IN_SECONDS);
+        // Since we are targeting a 1% slippage every 30 minutes the price range is defined as
+        // the price of a 1% move multipled by the amount of 30 second intervals in the auctionTimeToPivot
+        // This value is then divided by two to get half the price range
+        uint256 halfPriceRange = thirtyMinutePeriods.mul(onePercentSlippage).div(2);
+
+        // Auction start price is fair value minus half price range to center the auction at fair value
+        uint256 auctionStartPrice = fairValue.sub(halfPriceRange);
+        // Auction pivot price is fair value plus half price range to center the auction at fair value
+        uint256 auctionPivotPrice = fairValue.add(halfPriceRange);
+
+        return (auctionStartPrice, auctionPivotPrice);
+    }
+
+    /*
+     * Get USD value of one set
+     *
+     * @param  _btcPrice            The 18 decimal value of one full BTC
+     * @param  _ethPrice            The 18 decimal value of one full ETH
+     * @param  _naturalUnit         The naturalUnit of the set being valued
+     * @param  _units               The units of the set being valued
+     * @return uint256              The USD value of the set (in cents)
+     */
+    function calculateSetTokenPriceUSD(
+        uint256 _btcPrice,
+        uint256 _ethPrice,
+        uint256 _naturalUnit,
+        uint256[] _units
+    )
+        private
+        returns (uint256)
+    {
+        // Calculate btcDollarAmount of one Set Token (in cents) 
+        uint256 btcDollarAmount = calculateTokenAllocationAmountUSD(
+            _btcPrice,
+            _naturalUnit,
+            _units[0],
```

Comment:
Let's only reference these once and make sure they're consistent, e.g. passing in something more precise like `_bitcoinUnits` into the method. Fetch that once in `calculateNextSetUnits`